### PR TITLE
build: add travis continuous integration support

### PR DESCRIPTION
**Pull request by simon3z**

### CODE REVIEW

**Comment by simon3z** [comment]
@abonas tomorrow I'll add more info on why I made some of the changes. That's why I haven't sent the PR on friday.

File: lib/kubeclient.rb [hunk 1]
```diff
@@ -47,6 +47,8 @@ def get_resource_name(entity)
       end
     end
 
+    public
```
Comment (abonas) [inline lib/kubeclient.rb]: why the changes in this file are relevant for a travis ci pr?

Comment (simon3z) [inline lib/kubeclient.rb]: this fixes the get_/delete_/etc. visibility in ruby 2.1 (define_method scope is protected (2.1) vs public (2.0)... so I had to move this back here)

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+language: ruby
+rvm:
+  - 2.0
+  - 2.1
```
Comment (Fryguy) [inline .travis.yml]: Be sure to use strings for these.  Travis treats "2.0" to mean "the latest of Ruby 2.0" whereas using a float value does not do that.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+sudo: false
+cache: bundler
+script:
+  - bundle exec rake test rubocop
```
Comment (Fryguy) [inline .travis.yml]: why is this needed?  I think you can remove the script section entirely.

File: lib/kubeclient.rb (hunk 1 above)
Comment (Fryguy) [inline lib/kubeclient.rb]: This is what was causing me to fail locally on Ruby 2.1

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+language: ruby
+rvm:
+  - 2.0
+  - 2.1
+sudo: false
+cache: bundler
```
Comment (Fryguy) [inline .travis.yml]: I totally forgot about these 2...definitely good to have those in there.

File: .travis.yml (hunk 2 above)
Comment (simon3z) [inline .travis.yml]: It seems fine to me:
https://travis-ci.org/simon3z/kubeclient/builds/50911341

ruby 2.0.0p598 (2014-11-13 revision 48408) [x86_64-linux]

Comment (Fryguy) [inline .travis.yml]: If you look closely it says 2 instead of 2.0 for the  Ruby version.  This
is because of the String thing in the  yml.

File: .travis.yml (hunk 3 above)
Comment (simon3z) [inline .travis.yml]: Because travis was running only "test" and not "rubocop" (at least there's not trace of it in the console).

I'll fix the rake default target to execute rubocop as well.

File: .travis.yml (hunk 2 above)
Comment (simon3z) [inline .travis.yml]: If it's using latest 2.0, isn't it just cosmetic?

**Comment by simon3z** [comment]
@abonas @Fryguy this should be ok now.
